### ISSUE TITLE
graalvm 22.2.0+ breaks lib-only buildGraalvmNativeImage dependencies

### ISSUE THREAD

**Issue by hxtmdev** [NixOS/nixpkgs#214922]
### Describe the bug
Ever since dae7b421f7fd122dc5ce807be40ae9bd20ee78c1 (bisected) the optimized lib-only dependency mechanism from `buildGraalvmNativeImage` (see https://github.com/NixOS/nixpkgs/pull/147136) is not working anymore, resulting in huge closure sizes for `babashka` which make it unsuitable for inclusion in `streamLayeredImage`.

### Steps To Reproduce
1. Use `nixpkgs-unstable`
2. `nix-build '<nixpkgs>' -A babashka && nix path-info -S ./result -h`
3. Observe `/nix/store/4y5lr208saw6lk4yn2f4g5pkfg22b0h8-babashka-1.1.172       1.1G`

### Expected behavior
The mechanism from https://github.com/NixOS/nixpkgs/pull/147136 should continue to work.
```
$ cd nixpkgs
$ git checkout dae7b421f7fd122dc5ce807be40ae9bd20ee78c1~
$ nix-build . -A babashka && nix path-info -S ./result -h
/nix/store/8509cqvgkp1dh1vi3wy6fxvza0yj30kr-babashka-0.8.157     110.5M
```

### Additional context
`ldd` outputs
```
libpthread.so.0 => /nix/store/l7vp7c9z03dspbmss3gq5wdwx5c6ifcq-graalvm11-ce-22.3.0/lib/svm/clibraries/linux-amd64/libpthread.so.0 (0x000074a23d31e000)
```
instead of the previously working
```
libpthread.so.0 => /nix/store/5h3k50dcplzc4hpalhl61ih581la43dn-graalvm11-ce-22.1.0-lib/lib/libpthread.so.0 (0x0000794b70126000)
```
Attempts to insert `svm/clibraries/linux-amd64/` around https://github.com/NixOS/nixpkgs/blob/b5cff429fa31ac5a2e1d22fc1d3c164bf09a3330/pkgs/development/compilers/graalvm/community-edition/mkGraal.nix#L200 and https://github.com/NixOS/nixpkgs/blob/b5cff429fa31ac5a2e1d22fc1d3c164bf09a3330/pkgs/build-support/build-graalvm-native-image/default.nix#L13 were unsuccessful so far. Someone more familiar with those things should have a look at it.

### Notify maintainers
@thiagokokada

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.81-1.fc32.qubes.x86_64, Debian GNU/Linux, 11 (bullseye), nobuild`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(user): `"nixpkgs"`
 - nixpkgs: `/home/user/.nix-defexpr/channels/nixpkgs`

**Comment by thiagokokada** [comment]
It seems GraalVM is linking to its internal libraries again:

```
$ ldd ./result/bin/bb
	linux-vdso.so.1 (0x00007fff2669b000)
	libstdc++.so.6 => /nix/store/qbgfsaviwqi2p6jr7an1g2754sv3xqhn-gcc-11.3.0-lib/lib/libstdc++.so.6 (0x00007f77fc0cf000)
	libm.so.6 => /nix/store/l7vp7c9z03dspbmss3gq5wdwx5c6ifcq-graalvm11-ce-22.3.0/lib/svm/clibraries/linux-amd64/libm.so.6 (0x00007f77fbfef000)
	libpthread.so.0 => /nix/store/l7vp7c9z03dspbmss3gq5wdwx5c6ifcq-graalvm11-ce-22.3.0/lib/svm/clibraries/linux-amd64/libpthread.so.0 (0x00007f77fbfea000)
	libdl.so.2 => /nix/store/l7vp7c9z03dspbmss3gq5wdwx5c6ifcq-graalvm11-ce-22.3.0/lib/svm/clibraries/linux-amd64/libdl.so.2 (0x00007f77fbfe5000)
	librt.so.1 => /nix/store/l7vp7c9z03dspbmss3gq5wdwx5c6ifcq-graalvm11-ce-22.3.0/lib/svm/clibraries/linux-amd64/librt.so.1 (0x00007f77fbfde000)
	libc.so.6 => /nix/store/l7vp7c9z03dspbmss3gq5wdwx5c6ifcq-graalvm11-ce-22.3.0/lib/svm/clibraries/linux-amd64/libc.so.6 (0x00007f77fbdd5000)
	/nix/store/c35hf8g5b9vksadym9dbjrd6p2y11m8h-glibc-2.35-224/lib/ld-linux-x86-64.so.2 => /nix/store/9xfad3b5z4y00mzmk2wnn4900q0qmxns-glibc-2.35-224/lib64/ld-linux-x86-64.so.2 (0x00007f77fc2e7000)
	libgcc_s.so.1 => /nix/store/qbgfsaviwqi2p6jr7an1g2754sv3xqhn-gcc-11.3.0-lib/lib/libgcc_s.so.1 (0x00007f77fbdbb000)
```

I have a few ideas on how to possible workaround this issue, but this will probably need a full refactor of how we build the GraalVM Native Images and I don't exactly have time right now.

BTW, is this not a bug upstream? Because it looks like it is. Maybe we should report to them first.

**Comment by thiagokokada** [comment]
Created a PR: https://github.com/NixOS/nixpkgs/pull/215093

**Comment by hxtmdev** [comment]
Fixed in #215093
Thanks, @thiagokokada!
